src/haskell/install.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+set -e
+
+GHCUP_VERSION="0.1.19.2"
+GHCUP_BIN="x86_64-linux-ghcup-${GHCUP_VERSION}"
+BOOTSTRAP_HASKELL_GHC_VERSION="${VERSION:-"recommended "}"
+
+# Maybe install curl, gcc, make
+for x in curl gcc make; do
+    which $x > /dev/null || (apt update && apt install $x -y -qq)
+done
+
+GHCUP_DIR=~/.ghcup/bin
+mkdir -p $GHCUP_DIR
```

Comment:
I wonder if we should add a new user group `haskell` and chown & set sticky bit for this folder `.ghcup`

Ref: https://github.com/devcontainers/features/blob/main/src/ruby/install.sh#LL206-L209C7 &
https://github.com/devcontainers/features/blob/main/src/ruby/install.sh#L283-L285

Reply:
where 

> USERNAME="${USERNAME:-"${_REMOTE_USER:-"automatic"}"}"

We have https://github.com/devcontainers/features/blob/main/src/ruby/install.sh#L39-L59 to automatically figure out the USERNAME

Reply:
I'm not sure it's necessary, this will land in the home directory, but it doesn't really need a different username imho.